### ISSUE TITLE
Vervormde leafletkaartjes

### ISSUE THREAD

**Issue by RuttenAnneleen** [inbo/reporting-rshiny-grofwildjacht#431]
In de dev-versie van de faunabeheer website zijn de leafletkaartjes momenteel vervormd (uitgerokken en in bij wildschade ligt Vlaanderen in Frankrijk)

**Comment by SanderDevisscher** [comment]
issue occurs locally (via docker) & on UAT
On all public pages:
afschot :
![image](https://github.com/inbo/reporting-rshiny-grofwildjacht/assets/16779320/a2183e38-1d57-4113-9ff4-abd994ff059d)
wildschade :
![image](https://github.com/inbo/reporting-rshiny-grofwildjacht/assets/16779320/5225ac03-3334-45fe-a738-154916602c2c)
achtergrondinfo dashboard :
![image](https://github.com/inbo/reporting-rshiny-grofwildjacht/assets/16779320/41ec37df-e953-46cf-b272-ca941afb2ce2)

**Comment by SanderDevisscher** [comment]
issue is not present on private UAT (wbe-dev.inbo.be)

**Comment by SanderDevisscher** [comment]
This issue is breaking

**Comment by mvarewyck** [comment]
@SanderDevisscher I can reproduce. I guess it's due to a wrong projection. Did you update the spatial data?

**Comment by SanderDevisscher** [comment]
I did not consciously, but I'll check if something changed.

**Comment by mvarewyck** [comment]
> I did not consciously, but I'll check if something changed.

Last update is from 20th October: `spatialData_sf.RData 2023-10-20T09:42:20.000Z`

**Comment by SanderDevisscher** [comment]
I did update the spatial rdata but the source data for the communes for example did not change.

**Comment by SanderDevisscher** [comment]
In case of the wildschade maps, the points are placed correctly but the communes are placed somewhere in France<sup>1</sup>. 
But when I look at the source data for the communes they seem to be placed correctly<sup>2</sup>

<sup>1</sup>
![image](https://github.com/inbo/reporting-rshiny-grofwildjacht/assets/16779320/fd2cb1db-dbfa-43b1-9f2a-d99d67e0aac7)


<sup>2</sup>
![image](https://github.com/inbo/reporting-rshiny-grofwildjacht/assets/16779320/7eaf33e7-badc-4bff-9e95-57d6974eab82)

**Comment by mvarewyck** [comment]
@SanderDevisscher Could you send me one of the geojson files you use as input in `createShapeData()`? E.g. provinces.geojson.

**Comment by SanderDevisscher** [comment]
this zip contains provinces.geojson I use as input. this file was last changed 3 years ago.
[provinces.zip](https://github.com/inbo/reporting-rshiny-grofwildjacht/files/13113638/provinces.zip)

**Comment by mvarewyck** [comment]
The input file looks fine. I'm creating new shapes `spatialData_mv.RData` in the bucket, to detect whether the problem persists.

**Comment by mvarewyck** [comment]
If I create new shape data, the maps look correct
```
readS3(file = "spatialData_mv.RData")
leaflet(spatialData$provinces) %>% addTiles() %>% addPolygons()
```

If I use the current sf shapes on the bucket, it doesn't look fine
```
readS3(file = "spatialData_sf.RData")
leaflet(spatialData$provinces) %>% addTiles() %>% addPolygons()
```

@SanderDevisscher Which R package versions did you use when creating the shape data?
```
> packageVersion("geojsonsf")
[1] ‘2.0.3’
> packageVersion("sf")
[1] ‘0.9.8’
```

**Comment by SanderDevisscher** [comment]
I used the following versions:  
```
> packageVersion("geojsonsf")
[1] ‘2.0.3’
> packageVersion("sf")
[1] ‘1.0-14’
```

**Comment by SanderDevisscher** [comment]
so probably sf is the culprit

**Comment by mvarewyck** [comment]
> so probably sf is the culprit

Exactly, I found this in their [NEWS file](https://cran.r-project.org/web/packages/sf/news/news.html):

`use s2 spherical geometry as default when coordinates are ellipsoidal. This can be switched off (defaulting to planar geometry, using GEOS, as in sf < 1.0-0) by setting environment variable _SF_USE_S2 to false before package sf is loaded, or by sf_use_s2(FALSE); #1649`

I can try to implement a fix in the code.
For now, I think the quickest solution is you add `sf_use_s2(FALSE)` in your script?

**Comment by SanderDevisscher** [comment]
I should read the News more often :-/

**Comment by mvarewyck** [comment]
> I should read the News more often :-/

No one does, until they get into trouble :)

NB: I'm not allowed to remove files from S3 bucket anymore. So I failed to remove my test file
`aws.s3::delete_object(object = "spatialData_mv.RData", bucket = config::get("bucket", file = system.file("config.yml", package = "reportingGrofwild")))`

**Comment by SanderDevisscher** [comment]
I'll remove it

**Comment by SanderDevisscher** [comment]
PS I've uploaded the files using `sf_use_s2(FALSE)` but my laptop is busy with something else so haven't had time to test!

**Comment by mvarewyck** [comment]
The maps don't look fine yet. I'll implement a fix in the code on our side.

**Comment by SanderDevisscher** [comment]
I forgot to load sf so `sf_use_s2(FALSE)` failed, rerunning now

**Comment by SanderDevisscher** [comment]
done, testing locally

**Comment by SanderDevisscher** [comment]
issue persist :-(

**Comment by SanderDevisscher** [comment]
I've deleted "spatialData_mv.RData"

**Comment by mvarewyck** [comment]
> I've deleted "spatialData_mv.RData"

I see :) You'll have to do once more.

**Comment by SanderDevisscher** [comment]
> > I've deleted "spatialData_mv.RData"
> 
> I see :) You'll have to do once more.

done

**Comment by mvarewyck** [comment]
> should I upload the spatialdata again ?

Yes, I want to be sure it works on your side using `createShapeData()`

**Comment by SanderDevisscher** [comment]
[WBE_binnengrenzen_2023.zip](https://github.com/inbo/reporting-rshiny-grofwildjacht/files/13175147/WBE_binnengrenzen_2023.zip)
